File: src/countSquare.java
public class countSquare {
        public int countSquare(int[][] matrix) {
            int num = 0;
            int width = matrix[0].length;
            if(width == 0){
                return 0;
            }
            int height = matrix.length;
            if(height == 0){
                return 0;
            }
            int standard = width > height?width:height - 1;
            int tmp = 2;

            int result = 0;
            for(int i=0;i<height;i++){
                for(int j=0;j<width;j++){
                    //when square is single 1
                    if(matrix[i][j] == 1 && i>0 && j>0){
                        matrix[i][j] = 1+ Math.min(matrix[i-1][j-1],Math.min(matrix[i-1][j],matrix[i][j-1]));
                    }
                    num  += matrix[i][j];
                }


            }
            return num;
        }
}

File: src/countSquare.py
class Solution:
    def countSquares(self, matrix: List[List[int]]) -> int:
        num = 0

        height = len(matrix)
        width = len(matrix[0])

        dp = [[0 for i in range(width)] for j in range(height)]

        #edge
        for m in range(0, height):
            for n in range(0, width):
                #the single one on edge is not changed
                if m> 0 and n>0 and matrix[m][n] == 1:
                    matrix[m][n] += min(matrix[m-1][n-1], min(matrix[m][n-1], matrix[m-1][n]))
                num += matrix[m][n]
        return num
